addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/HAConfiguration.java

```diff
@@ -21,16 +21,64 @@
 import org.eclipse.smarthome.core.thing.Thing;
 
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import com.google.gson.annotations.SerializedName;
 
 /**
  * Base class for home assistant configurations.
  *
  * @author Jochen Klein - Initial contribution
  */
+
 @NonNullByDefault
 public abstract class HAConfiguration {
 
+    public static final Factory FACTORY = new Factory();
```

Comment:
Another singleton. I rather see the gson field as a WeakReference static field variable, injected by the ThingFactory (as that class ist our OSGi started object). That way it can be garbage collected as soon as the ThingFactory gets disposed.

Reply:
I kind of hate, that we have to pass a parameter (gson) through so many other calls, which already have (too) many parameter. Do I tried to avoid it.
Would this work:
```    @Nullable
    private static WeakReference<Factory> FACTORY = null;

    public static Factory instance() {
        synchronized (HAConfiguration.class) {
            @Nullable
            Factory result = null;

            if (FACTORY != null) {
                result = FACTORY.get();
            }
            if (result == null) {
                result = new Factory();
                FACTORY = new WeakReference<HAConfiguration.Factory>(result);
            }
            return result;
        }
    }
```

Reply:
Another idea: Can we turn CFactory into a builder-pattern and then pass the builder in the contrutor to the components/AbstractComponent.
That ways to do not have the repeative long parameter list
e.g.
```
CFactory.build().withThingUID(..).withHaID(...).withConfig(..).withListener(..).withGson(..).create();
```
and then
```
AbstractComponent(CFactory.Builder builder) {
   ...
   this.configJson = builder.getConfig();
   ...
}
````